packages/flutter/lib/src/gestures/mouse_tracking.dart

```diff
@@ -556,26 +591,31 @@ class MouseTracker extends ChangeNotifier {
   ///    This is to prevent exceptions caused calling setState of a disposed
   ///    widget. See [MouseTrackerAnnotation.onExit] for more details.
   ///  * The [MouseTracker] also uses the attachment to track the number of
-  ///    attached annotations, and will skip mouse position checks if there is no
+  ///    attached annotations, and will skip mouse position checks if there are no
   ///    annotations attached.
   /// {@endtemplate}
   ///  * Attaching an annotation that has been attached will assert.
-  void attachAnnotation(MouseTrackerAnnotation annotation) {
+  void attachAnnotation(Key annotationKey) {
```

Comment:
SHould this continue to tale an annotation and just internally grab the key from there?

Reply:
I've tried to keep it `MouseTrackerAnnotation` for a long time during the development of this PR, but eventually removed it because it makes the implementation of `RenderMouseRegion` noticeably more complicated.

One of the problems is that, when the render object is detached, it needs to detach the current annotation. If `detachAnnotation` only accepts `MouseTrackerAnnotation`s, then either the render object must keep its own hover annotation, or it builds a fake annotation when disposed. Keeping the hover annotation adds quite a few bookkeeping overhead, since otherwise it could've built the annotation anew during every `paint` (which is how the PR is implemented currently, see [here](https://github.com/flutter/flutter/pull/46034/files/7f4a05dac5ae93bacb1a353f7a194c4083eb4d06#diff-f41d041ddd3fe665cc9e123f992bb5e0R2809)).